server/src/main/java/com/flipkart/varadhi/web/v1/produce/ProduceHandlers.java

```diff
@@ -89,8 +97,8 @@ public void produce(RoutingContext ctx) {
                                     ctx.endRequestWithResponse(produceResult.getMessageId());
                                 } else {
                                     ctx.endRequestWithStatusAndErrorMsg(
-                                            getHttpStatusForProduceStatus(produceResult.getProduceStatus().status()),
```

Comment:
what happens if we dont run this whenComplete handler using runOnContext()?

Reply:
Not sure, but I believe it should be fine as long as context is maintained. However, i guess runOnContext() or ctx.end() both seems to be ending up on event loop thread.